core-backup/src/main/scala/io/aiven/guardian/kafka/backup/BackupClientInterface.scala

```diff
@@ -369,9 +412,16 @@ object BackupClientInterface {
     }
 
   protected def calculateNumberOfPeriodsFromTimestamp(initialTime: OffsetDateTime,
-                                                      period: FiniteDuration,
+                                                      timeConfiguration: TimeConfiguration,
                                                       reducedConsumerRecord: ReducedConsumerRecord
-  ): Long =
+  ): Long = {
+    val (period, finalInitialTime) = timeConfiguration match {
+      case PeriodFromFirst(duration) => (duration, initialTime)
+      case ChronoUnitSlice(chronoUnit) =>
```

Comment:
If its configured for the suspend/resume case then we need to set the initial time to also be the start of the time unit